accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,188 @@
+        blocklyApp.FieldView],
+    inputs: ['block', 'displayBlockMenu', 'level', 'index', 'tree'],
+  })
+  .Class({
+    constructor: [blocklyApp.ClipboardService, blocklyApp.TreeService,
+                  function(_clipboardService, _treeService) {
+      this.infoBlocks = {};
+      this.nextBlock = {};
+      this.sharedClipboardService = _clipboardService;
+      this.treeService = _treeService;
+    }],
+    setLabelledBy: function(item, id) {
+      if (!item.getAttribute('aria-labelledby')) {
+        item.setAttribute('aria-labelledby', id);
+      }
+    },
+    concatStringWithSpaces: function() {
+      var string = arguments[0];
+      for (i = 1; i < arguments.length; i++) {
+        var arg = arguments[i];
+        // arg can be undefined and if so, should be ignored.
+        if (arg) {
+          string = string + ' ' + arguments[i];
+        }
+      }
+      return string;
+    },
+    createCategoryDependantId: function(index, parentList) {
+      // if this is the first block in a category-less toolbox,
+      // the id should be blockly-toolbox-tree-node0
+      if (index !== undefined && index === 0) {
+        if (this.tree.getAttribute('aria-activedescendant') ==
+          'blockly-toolbox-tree-node0') {
+          this.addClass(parentList, 'blocklyActiveDescendant');
+          parentList.setAttribute('aria-selected', 'true');
+          this.treeService.setActiveDesc(parentList, this.tree.id);
+        }
+        return 'blockly-toolbox-tree-node0';
+      } else {
+        return this.treeService.createId(parentList);
+      }
+    },
+    addClass: function(node, classText) {
+      // ensure that node doesn't have class already in it
+      var classList = node.className;
+      classList = classList.split(' ');
+      var canAdd = true;
+      for (var i = 0; i < classList.length; i++) {
+        if (classList[i] == classText) {
+          canAdd = false;
+        }
+      }
+      // add class if it doesn't
+      if (canAdd) {
+        if (classList.length) {
+          node.className += classText;
+        } else {
+          node.className += ' ' + classText;
+        }
+      }
+    },
+    getInfo: function(block) {
+      // list all inputs
+      if (this.infoBlocks[block.id]) {
+        this.infoBlocks[block.id].length = 0;
+      } else {
+        this.infoBlocks[block.id] = [];
+      }
+
+      var blockInfoList = this.infoBlocks[block.id];
+
+      for (var j = 0, field; field = block.fieldRow[j]; j++) {
+        blockInfoList.push(field);
+      }
+
+      return this.infoBlocks[block.id];
+    },
+    getInputTypeLabel: function(connection) {
+      if (connection.check_) {
+        return connection.check_.join(', ').toUpperCase();
+      } else {
+        return 'any';
+      }
+    },
+    copyToWorkspace: function(block) {
+      var xml = Blockly.Xml.blockToDom(block);
+      Blockly.Xml.domToBlock(blocklyApp.workspace, xml);
+      alert('block added to workspace');
```

Comment:
For now just create a messages.js in the accessibility folder.  We can deal
with translations later.

Send (awkwardly) from an Android phone.
On 6 May 2016 12:40 p.m., "M" <notifications@github.com> wrote:

In accessible/toolbox_treeview.component.js
<https://github.com/google/blockly/pull/354#discussion_r62380619>:

> +        blockInfoList.push(field);
> +      }
> +
> +      return this.infoBlocks[block.id];
> +    },
> +    getInputTypeLabel: function(connection) {
> +      if (connection.check_) {
> +        return connection.check_.join(', ').toUpperCase();
> +      } else {
> +        return 'any';
> +      }
> +    },
> +    copyToWorkspace: function(block) {
> +      var xml = Blockly.Xml.blockToDom(block);
> +      Blockly.Xml.domToBlock(blocklyApp.workspace, xml);
> +      alert('block added to workspace');

I agree that all strings will need to be pulled out. Do I make a file in
the msg folder and put them all there? @NeilFraser
<https://github.com/NeilFraser>

—
You are receiving this because you were mentioned.

Reply to this email directly or view it on GitHub
<https://github.com/google/blockly/pull/354/files/f62f55233385888659690fa32de4854ef92e732e#r62380619>